browserApp/collection_view.py

```diff
@@ -9,54 +14,64 @@ def __init__(self, parent=None):
 
         layout = QtWidgets.QHBoxLayout()
 
-        self.label = QtWidgets.QLabel()
-        self.label.setText("Collection")
+        #self.label = QtWidgets.QLabel()
+        self.labels = []
 
         layout.addSpacing(1)
-        layout.addWidget(self.label)
+        #layout.addWidget(self.label)
         layout.addSpacing(1)
 
         self.setLayout(layout)
 
     def populate(self, items):
-        self.label.setText("")
+        #self.label.clear()
+        for label in self.labels:
+            label.clear()
+            self.layout().removeWidget(label)
```

Comment:
See this post regarding removing items from a layout - it's actually a bit tricky in pyqt
https://stackoverflow.com/questions/9374063/remove-all-items-from-a-layout